lib/WWW/YouTube/Download.pm

```diff
@@ -186,6 +195,34 @@ sub prepare_download {
     };
 }
 
+sub _mime_to_suffix {
+      my $mime = shift;
+      return $mime =~ m{(?:video|audio)/webm;} ? 'webm'
+           : $mime =~ m{(?:video|audio)/mp4;}  ? 'mp4'
+           : $mime =~ m{video/3gp;}            ? '3gp'
+           :                                     'flv'
+      ;
+}
+
+sub _decode_player_response {
```

Comment:
Would it be possible to add a basic unit test for this sub?

Reply:
Added by mocking `_get_content`.